### ISSUE TITLE
CLI option to start a player and exit immediately for pass-though streams

### ISSUE THREAD

**Issue by Tomalak** [streamlink/streamlink#2535]
## Feature Request

- [x] This is a feature request and I have read the contribution guidelines.

### Description

Assume we start an pass-through stream, for example, Twitch VOD via HLS handed over to VLC. 

All that Streamlink really does in this situation is handle the communication with the video source to figure out the actual media URL before starting the configured player.

After that, the player acts independently and Steamlink sits around idle, as the stream itself does not go through it at all, contrary to what the word "pass-though" suggests.

### Expected / Actual behavior

 - **actual behavior** - the Streamlink process never exits on its own
 - **desirable behavior** - a command line switch exisits that tells Streamlink to shut down after it has successfully launched the player

### Additional comments, screenshots, etc.

How I noticed: I'm using Streamlink on Windows. For convenience, I have a small batch file that collects a video URL from the clipboard and starts Streamlink with it. Once VLC runs, the console window hosting Streamlink hangs around indefinitely, but it can also be closed without any ill-effects.

This is somewhat annoying, as useless console windows start stacking up after watching a few videos. A command line option to shut down Streamlink cleanly on its own would solve this elegantly. 

It also uses system resources when it does not really need to. The idle Console Host and Python processes eat about 40 MB on my machine for each such window.

**Comment by bastimeyer** [comment]
You can use Streamlink's [`--stream-url` parameter](https://streamlink.github.io/cli.html#cmdoption-stream-url) for this and use the output as the player's input parameter.

Example in BASH:
```bash
player --foo --bar=baz "$(streamlink --stream-url URL QUALITY)"

# or

INPUT="$(streamlink --stream-url URL QUALITY)"
player --foo --bar=baz "${INPUT}"
```

**Comment by Tomalak** [comment]
Hmmm, that would be possible. I haven't thought about it this way. Unfortunately this makes it harder to see what's going on, since all the usual output from Streamlink is suppressed.

In any case, there is no real reason for Streamlink to stay open when it explicitly has nothing to do. From that perspective it might still be worth consideration.

**Comment by bastimeyer** [comment]
No, Streamlink not terminating its process when using the passthrough option does have a reason. It may be annoying when using external scripts, but if it would terminate, then the user would have no chance closing the spawned player childprocess from the shell by sending a SIGINT or SIGTERM signal (eg. CTRL+c).

The lack of output when using `--stream-url` however is a problem. In this case, Streamlink should output its log to `stderr`, just like when using the `--stdout` option.

**Comment by Tomalak** [comment]
...and that's where an option to tell it that it can safely terminate would come in handy. :)

**Comment by beardypig** [comment]
Seems like this might be a Windows specific request, in Linux and macOS Streamlink controla the player process - if you kill streamlink it will kill the player (I think this is the case with pass-through). We could probably add a `player-detached` option or something, but if there is a lot of operating system specific code it might not be worth the maintenance hassle. With regards to the log going to `stderr` for the `stream-url` option, we should look in to that, not sure if Windows she’ll handles things differently though...

**Comment by beardypig** [comment]
@Tomalak does using `streamlinkw` instead of `streamlink` help? I think it might depend on how you launch it though, I don't use Windows so I cannot say wether it would work for sure or not, @bastimeyer might know :)

**Comment by skulblakka** [comment]
>in Linux and macOS Streamlink controla the player process

I think the problem is that streamlink does not handle the user closing the command prompt window directly using the "X" button causing the player to stay open (with and without using ```--player-passthrough```). 

I also noticed that streamlink does not handle the Keyboard Interrupt while using ```--player-passthrough```. Instead it will handle it when the player is closed manually.

**Comment by beardypig** [comment]
Does the player close if you are not using `passthrough` and close the command prompt with `X`? 
I'm not very familiar with how Windows works with command prompts etc (I haven't used Windows in earnest for years), but I'm not sure that Streamlink would be able to detect that the command prompt has been closed, as I think the Python process would be terminated without chance to clean up...

**Comment by gravyboat** [comment]
@beardypig Yes it does. I'm on Windows 10 and my player always closes if I simply close the command prompt or powershell window as long as I don't use `--player-passthrough` which makes sense since `passthrough` would be bypassing Streamlink so it no longer matters if you close the window. If people really want to do this they can just start it in a Powershell window like this:
```
streamlink --player-passthrough hls https://www.twitch.tv/beyondthesummit; stop-process -Id $PID
```
Then when the player is closed the Powershell window will exit. There's no need for us to spend time on this.

**Comment by skulblakka** [comment]
>Yes it does.

As far as I can see this is not true. The player will not be terminated when you close the powershell window. Are you maybe using a player that closes automatically when it reaches the end like mpv (by default at least)?

Example:
```
streamlink twitch.tv/lirik best --player-args="--keep-open=yes {filename}"
```
Closing the powershell window now will make mpv pause and stay open instead of being terminated.

**Comment by gravyboat** [comment]
@skulblakka Ahh I'm not using `--keep-open` when I do this.

**Comment by CrendKing** [comment]
For whoever wants this with the streamlink output, here is another Windows script:
```
<streamlink command line>
timeout 10 > nul
taskkill /f /im streamlink.exe
```
This assumes the streamlink -> player transition is done within 10 seconds. You can increase that timeout as needed. Once the transition is complete, streamlink process can be terminated.

If we decide to let this happen, simplest change would be switch `subprocess.call` at https://github.com/streamlink/streamlink/blob/5075a5e1c7d21eb20677a9614345d5cf411f3c13/src/streamlink_cli/output.py#L235 to `subprocess.Popen`.
